bacula_server_operator/src/charm.py

```diff
@@ -0,0 +1,326 @@
+                "fd-password": secrets.token_urlsafe(32),
+                "sd-password": secrets.token_urlsafe(32),
+                "api-password": secrets.token_urlsafe(32),
+            }
+            secret = self.app.add_secret(content=passwords)
+            data["passwords"] = secret.id
+            return passwords
+        else:
+            secret = self.model.get_secret(id=password_secret_id)
+            return secret.get_content(refresh=True)
+
+    def _get_bacula_config(self) -> bacula.BaculaConfig:
+        """Get bacula configuration.
+
+        Returns:
+            a bacula configuration object.
+        """
+        passwords = self._get_peer_data()
+        return bacula.BaculaConfig(
+            dir_address=self._get_unit_address(),
+            sd_address=self._get_unit_address(),
+            dir_password=passwords["dir-password"],
+            fd_password=passwords["fd-password"],
+            sd_password=passwords["sd-password"],
+            file_retention=self.config.get("file-retention"),
+            job_retention=self.config.get("job-retention"),
+            volume_retention=self.config.get("volume-retention"),
+        )
+
+    def _get_baculum_api_config(self) -> bacula.BaculumApiConfig:
+        """Get baculum api configuration.
+
+        Returns:
+            a baculum api configuration object.
+        """
+        return bacula.BaculumApiConfig(
+            username=BACULUM_CHARM_USERNAME,
+            password=self._get_peer_data()["api-password"],
+        )
+
+    def _is_singleton(self) -> bool:
+        """Check if there are more than one unit in the bacula-server charm.
+
+        Returns:
+            True if there are only one unit in the bacula-server charm.
+        """
+        relation = self.model.get_relation(PEER_RELATION_NAME)
+        if not relation:
+            return False
+        unit_names = [unit.name for unit in relation.units]
+        if not unit_names:  # only one unit
+            return True
+        unit_names.sort(key=lambda name: int(name.split("/")[-1]))
+        return self.unit.name == unit_names[0]
+
+    def _list_relation_fd(self) -> list[relations.BaculaFdInfo]:
+        """List all established bacula-dir relations.
+
+        Returns:
+            a list of bacula-dir relations.
+        """
+        self._bacula_dir_relation.send_to_bacula_fd()
+        return self._bacula_dir_relation.receive_from_bacula_fd()
+
+    def _reconcile(self) -> None:
+        """Reconcile the bacula-server."""
+        bacula_config = self._get_bacula_config()
+        baculum_api_config = self._get_baculum_api_config()
+        db_config = self._get_db_config()
+        s3_config = self._get_s3_config()
+        if not self._is_singleton():
+            raise UnrecoverableError("bacula-server charm does not support multiple units")
+        if not self._bacula.is_installed():
+            self.unit.status = ops.MaintenanceStatus("installing charmed-bacula-server")
```

Comment:
Shouldn't the status be changed after calling `install`?

Reply:
Added, thanks!